### PR TITLE
Logging config sources on startup

### PR DESCRIPTION
**Pull request by hem-brahmbhatt**
#### Background

We've struggled in the past to see what config has been loaded and has been the source of failing builds.

This change makes it easier to see what config has been loaded during the following processes:
- `yarn server`
- `yarn test`
- `node app.js` which is used to start the server across our environments

We should see logs containing the config files loaded, e.g.
screen shot 2018-11-27 at 16 18 32

#### How has this been tested?

Please indicate the need for this change to be tested. If not please justify, if so then at what levels. **Reviewers** ensure that these test requirements are also reviewed.

- [x] Unit / Integration tests
- [ ] Browser tests
- [ ] End2end tests

### CODE REVIEW

**Comment by diversemix** [comment]
Which issue does this relate to?

/cc @hdrury1

File: package.json [hunk 1]
```diff
@@ -35,6 +35,7 @@
     "start:styleguide": "docker-compose run --no-deps -p 6060:6060 app yarn run styleguide",
     "stop:services": "docker-compose down -v",
     "styleguide": "styleguidist server",
+    "pretest": "NODE_ENV=test log-config-sources",
```
Comment (giorgiosironi) [inline package.json]: can you post a sample output of this command?

Comment (giorgiosironi) [reply package.json]: (sample log, more than output)

File: server/xpub-server/index.js [hunk 2]
```diff
@@ -1,5 +1,9 @@
 const { merge } = require('lodash')
 const fs = require('fs')
+const logger = require('@pubsweet/logger')
+const { logConfigSources } = require('@elifesciences/util')
+
+logConfigSources(logger.info)
```
Comment (giorgiosironi) [inline server/xpub-server/index.js]: reasonable log level

File: package.json (hunk 1 above)
Comment (giorgiosironi) [reply package.json]: ah it's using `console.log` in this case, so actual output

File: server/util/package.json [hunk 3]
```diff
@@ -0,0 +1,12 @@
+{
+  "name": "@elifesciences/util",
+  "version": "1.0.0",
+  "main": "index.js",
+  "license": "ISC",
```
Comment (diversemix) [inline server/util/package.json]: If we were to release this we need to change the license

**Comment by damnhipster** [comment]
Closing until after initial release.

**Pull request closed without merging**